src/main/java/ch/uzh/ifi/hase/soprafs22/entity/BlockedUserRelation.java

```diff
@@ -0,0 +1,76 @@
+package ch.uzh.ifi.hase.soprafs22.entity;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.io.Serializable;
+import java.util.*;
+
+/**
+ * This Entity Class is used as a relationship class that holds the relationship if a user blocks another user. Basically,
+ * it works exactly like Matches, therefore, if a user deletes his / her profile, the relationship gets deleted on
+ * cascade. It creates two tables: One with the id and the creationDate and one with the relationship between the users.
+ */
+@Entity
+@Table(name = "BLOCK_RELATION")
+public class BlockedUserRelation implements Serializable {
+  @Id
+  @GeneratedValue
+  @Column(name = "id")
+  private long blockedRelationId;
+
+  // In order that the block_relation table does not only hold the id ...
+  @Column(name = "creation_date")
+  private final Date creationDate = new Date();
+
+  @ManyToMany(fetch = FetchType.LAZY,
+          cascade = {CascadeType.DETACH, CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REFRESH})
+  @JoinTable(name = "BLOCKED_USER_RELATION",
+          joinColumns = {@JoinColumn(referencedColumnName = "id")},
+          inverseJoinColumns = {@JoinColumn(referencedColumnName = "user_id")})
```

Comment:
instead of id: blockedUserRelation_id?
makes it insteantly clear which id is meant. but only nice to have

Reply:
First had it like this, but tbh., we never need the name in the server, but it only constitutes the name of the columns in the tables. But for JoinTable, hibernate concatenates "BLOCKED_USER_RELATION" with "id", so the column name would become very long: "BLOCKED_USER_RELATION_BLOCKEDUSERRELATION_ID".

Reply:
![image](https://user-images.githubusercontent.com/75019468/169900617-5e1617ae-e4e9-440a-8952-5574b072e5b0.png)

Reply:
Instead, it would it make even more clear to have "id" for all id's instead of user_id, match_id, etc.

Reply:
yes, i remember now. OneToMany was the same